### PR TITLE
round traits framework + a few placeholders

### PR DESCRIPTION
**Pull request by harryob**
this allows for us to have randomised traits that will impact a whole round for a bit of variety. wouldn't that be nice?

:cl:
add: added the framework for random traits impacting the whole round
add: added a number of traits, including flu season, faulty piping, book balancing, illicit dealings, faulty cryogenics, economic boom/slump, temperature change and mismatched squads
/:cl:

these should (if i've done the weighting right) not appear enough to be annoying

also

# important

i would like to add more, especially for xenos and i'll be going through some ideas i curated prior to merge

### CODE REVIEW

File: code/datums/round_traits/neutral.dm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/datum/round_trait/black_market/on_round_start()
+	supply_controller.black_market_enabled = TRUE
+
+/datum/round_trait/bad_wakeup
+	name = "Faulty Cryogenics"
+	trait_type = ROUND_TRAIT_NEUTRAL
+	weight = 5
+	show_in_human_report = TRUE
+	human_report_message = "ARES was flagging up some errors for a minor subsystem in the cryogenics pods. Should all be fine."
```
Comment (carlarctg) [inline code/datums/round_traits/neutral.dm:16]:
```suggestion
	human_report_message = "ARES was flagging up some errors for a minor subsystem in the cryogenics pods. Should be all fine."
```

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Seemingly non-functional during tests on live.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/datums/round_traits/neutral.dm [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	trait_to_give = TRAIT_ROUND_CRYOSLEEP_SICKNESS
+
+/datum/round_trait/economic_boom
+	name = "Economic Boom"
+	trait_type = ROUND_TRAIT_NEUTRAL
+	weight = 5
+	show_in_human_report = TRUE
+	human_report_message = "A small bonus has been deposited in the accounts of our personnel by The Company. Seemingly, they want everyone to co-operate with them on this operation."
+	trait_to_give = TRAIT_ROUND_ECONOMIC_BOOM
+	blacklist = list(/datum/round_trait/economic_boom)
```
Comment (Zonespace27) [inline code/datums/round_traits/neutral.dm:26]: This should be slump

File: code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm [hunk 3]
```diff
@@ -15,6 +15,32 @@
 	return GLOB.not_incapacitated_and_adjacent_strict_state
 
 /obj/structure/machinery/cm_vending/sorted/cargo_guns/squad_prep/populate_product_list(scale)
+	if(HAS_TRAIT(SSround, TRAIT_ROUND_STANDARD_ISSUE))
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm:18]: Is this used?

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Pull request closed without merging**